src/main/java/javaslang/collection/List.java

```diff
@@ -133,13 +136,19 @@
             @SuppressWarnings("unchecked")
             final List<T> list = (List<T>) elements;
             return list;
-        } else if (elements instanceof ArrayList || elements instanceof Vector) {
-            @SuppressWarnings("unchecked")
-            final java.util.List<T> indexedList = (java.util.List<T>) elements;
+        } else if (elements instanceof java.util.List<?>) {
```

Comment:
Reflexes are a good thing - I also do that in different contexts. In general Java's type system is funny (sort of). IMO for example the assignment `List<T> list2 = list` should be valid, where `list` is of type `List<? extends T>`, but it does not compile. I do that in some locations of Javaslang and suppress the 'unchecked' warnings...